packages/aws-cdk-lib/custom-resource-config/test/custom-resource-config.test.ts

```diff
@@ -0,0 +1,547 @@
+                    Ref: Match.stringLikeRegexp('AWS')
+                }
+            }
+        })
+    })
+    test('awsapi - snapshot', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        const connection = new events.Connection(stack, "Connection", {
+            authorization: events.Authorization.apiKey(
+                "x-api-key",
+                cdk.SecretValue.unsafePlainText("no-secret")
+            ),
+            description: "Connection with API Key x-api-key",
+        });
+        const destination = new events.ApiDestination(stack, "Destination", {
+            connection,
+            endpoint: "https://amazon.com",
+            description: "Calling example.com with API key x-api-key",
+        });
+        const rule = new events.Rule(stack, "Rule", {
+            schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+            targets: [new targets.ApiDestination(destination)],
+        });
+        const awsApi = new targets.AwsApi({
+            action: "action",
+            service: "service",
+            apiVersion: "apiVersion",
+            catchErrorPattern: "catchErrorPattern",
+            parameters: "hello-world",
+            policyStatement: new iam.PolicyStatement({
+                actions: ["s3:GetObject"],
+                resources: ["*"],
+            }),
+        });
+        rule.addTarget(awsApi);
+
+        const template = Template.fromStack(stack);
+
+        expect(template.toJSON()).toMatchSnapshot();
+    })
+})
+
+describe('LogGroup', () => {
+    test('s3LogGroup - UnModified by Aspects', () => {
+        const LOCALLY_SET_LOG_RETENTION = logs.RetentionDays.ONE_WEEK;
+
+        const app = new cdk.App();
+        const stack = new cdk.Stack(app);
+
+        const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+        new s3deploy.BucketDeployment(stack, 's3deployLogGroup', {
+            sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+            destinationBucket: websiteBucket,
+            logGroup: new logs.LogGroup(stack, 'LogGroup', {
+                retention: LOCALLY_SET_LOG_RETENTION,
+            }),
+        });
+
+        const template = Template.fromStack(stack);
+
+        template.resourceCountIs('AWS::Logs::LogGroup', 1);
+        template.hasResourceProperties('AWS::Logs::LogGroup', {
+            RetentionInDays: LOCALLY_SET_LOG_RETENTION,
+        });
+        template.hasResourceProperties('AWS::Lambda::Function', {
+            LoggingConfig: {
+                LogGroup: {
+                    Ref: Match.stringLikeRegexp('LogGroup')
+                }
+            }
+        })
+    });
+
+    test('s3LogGroup - update lambda logGroup to SET_LOG_RETENTION', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
```

Comment:
Most constant variables should not be all caps. The only constants that we use this naming convention for are variables that hold some magic value we don't want to typo, or that would otherwise be difficult to read.